addons/address.py

```diff
@@ -202,6 +202,23 @@ def _inet_address_list_config(self, ifaceobj, newaddrs, newaddr_attrs):
             except Exception, e:
                 self.log_error(str(e), ifaceobj)
 
+    def _enforce_cidr_notation(self, addr_list):
+        addr_list_cidr = []
+
+        if not addr_list:
+            return None
+
+        for addr in addr_list:
+            addr_cidr = addr
+
+            if not '/' in addr:
+               plen = '/128' if ':' in addr else '/32'
+               addr_cidr += plen
+
+            addr_list_cidr.append(addr_cidr)
+
+        return addr_list_cidr
+
```

Comment:
You should use the IPNetwork object instead, something like:
```suggestion
for addr in addr_list:
    try:
        addr_list_cidr.append(str(IPNetwork(addr)))
    except:
        pass # maybe log.warning here
```

Reply:
If I append only the str()-ified version there's no difference, it's still a string in the list. What would that again besides burning CPU cycles for parsing and stringifying the IP?

Reply:
At that point in the code has the address been validated? IPNetwork would ensure that we have a valid ip address here, otherwise it would raise an exception with an appropriate error message.

I understand the concern about excessive code but i'm trying to move ifupdown2 code base to use the ipaddr module more often and stop doing manual checkings everywhere. Thats why i'm not a big fan of adding this like that.

In the future, IP address should only be handled with IPNetwork objects and not strings.

Reply:
`runningaddrs = utils.get_normalized_ip_addr(ifaceobj.name, self._enforce_cidr_notation(addrs_read))`

`get_normalized_ip_addr(cls, ifacename, ipaddrs)` does this. But I didn't want to add the host-netmask there as I don't know if this a wise idea to add this globally.